# how many passengers 18 and up had a kid or a spouse

# i plan on doing a retrospective observational study
# i will do a census of the data because i need every person above 18 and not a random selection
# a bias that could be in my data is that some people were never recorded as having a kid or spouse in this dataset because of stowaways or people wanting to keep their personal life to themself.

In [1]:
import pandas as pd

print(f"Pandas version: {pd.__version__}")

Pandas version: 2.1.4


In [2]:
df = pd.read_csv('titanic.csv')

In [3]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 891 entries, 0 to 890
Data columns (total 12 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   PassengerId  891 non-null    int64  
 1   Survived     891 non-null    int64  
 2   Pclass       891 non-null    int64  
 3   Name         891 non-null    object 
 4   Sex          891 non-null    object 
 5   Age          714 non-null    float64
 6   SibSp        891 non-null    int64  
 7   Parch        891 non-null    int64  
 8   Ticket       891 non-null    object 
 9   Fare         891 non-null    float64
 10  Cabin        204 non-null    object 
 11  Embarked     889 non-null    object 
dtypes: float64(2), int64(5), object(5)
memory usage: 83.7+ KB


In [4]:
num_samples = len(df)
print(num_samples)

891


In [5]:
Row_Pos = df.iloc[7]
print(Row_Pos)

PassengerId                                 8
Survived                                    0
Pclass                                      3
Name           Palsson, Master. Gosta Leonard
Sex                                      male
Age                                       2.0
SibSp                                       3
Parch                                       1
Ticket                                 349909
Fare                                   21.075
Cabin                                     NaN
Embarked                                    S
Name: 7, dtype: object


In [6]:
df[df["Embarked"] == "Q"]

PassengerId  Survived  Pclass                                  Name  \
5              6         0       3                      Moran, Mr. James   
16            17         0       3                  Rice, Master. Eugene   
22            23         1       3           McGowan, Miss. Anna "Annie"   
28            29         1       3         O'Dwyer, Miss. Ellen "Nellie"   
32            33         1       3              Glynn, Miss. Mary Agatha   
..           ...       ...     ...                                   ...   
790          791         0       3              Keane, Mr. Andrew "Andy"   
825          826         0       3                       Flynn, Mr. John   
828          829         1       3          McCormack, Mr. Thomas Joseph   
885          886         0       3  Rice, Mrs. William (Margaret Norton)   
890          891         0       3                   Dooley, Mr. Patrick   

        Sex   Age  SibSp  Parch  Ticket     Fare Cabin Embarked  
5      male   NaN      0      0  330877   8.4583   NaN        Q  
16     male   2.0      4      1  382652  29.1250   NaN        Q  
22   female  15.0      0      0  330923   8.0292   NaN        Q  
28   female   NaN      0      0  330959   7.8792   NaN        Q  
32   female   NaN      0      0  335677   7.7500   NaN        Q  
..      ...   ...    ...    ...     ...      ...   ...      ...  
790    male   NaN      0      0   12460   7.7500   NaN        Q  
825    male   NaN      0      0  368323   6.9500   NaN        Q  
828    male   NaN      0      0  367228   7.7500   NaN        Q  
885  female  39.0      0      5  382652  29.1250   NaN        Q  
890    male  32.0      0      0  370376   7.7500   NaN        Q  

[77 rows x 12 columns]

In [7]:
count = 1
for i in range(len(df[(df["SibSp"] >= 1) & (df["Age"] >= 18)])):
    count += 1
print("the amount of people with a kid or a spouse above 18 is",count)
percent = round((count/num_samples)*100)
print("which is",percent,"% of people on board")

the amount of people with a kid or a spouse above 18 is 174
which is 20 % of people on board


# the amount of people with a kid or a spouse above 18 is 174 people, this is 20% of the people on board the titanic (in my dataset)